cmake/PkgConfig.cmake

```diff
@@ -1,3 +1,46 @@
-string(REGEX REPLACE " [^ ]*valhalla[^ ]+" "" deplibs "${deplibs}")
-string(REGEX REPLACE " [^ ]*lib([^/ ]+).so" " -l\\1" deplibs "${deplibs}")
-configure_file(${INPUT} ${OUTPUT} @ONLY)
+function(set_variable_from_rel_or_absolute_path var root rel_or_abs_path)
+  if(IS_ABSOLUTE "${rel_or_abs_path}")
+    set(${var} "${rel_or_abs_path}" PARENT_SCOPE)
+  else()
+    set(${var} "${root}/${rel_or_abs_path}" PARENT_SCOPE)
+  endif()
+endfunction()
+
+# configure a pkg-config file libvalhalla.pc
+function(configure_proj_pc)
```

Comment:
this function is called from the root CMakeLists.txt and sets up the libvalhalla.pc properly depending on the build options.